### PR TITLE
Feature/30 service to controller

### PR DESCRIPTION
**Pull request by dawciobiel**
30 - Create enum service to controller status codes.
31 - Create DTO class to share information between Services and Controllers.

### CODE REVIEW

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/dto/RscpDTO.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ * {@code RscpDTO} record used to transfer date between service and controller. Contain a status, message and
+ * body.
+ *
+ * @param status  the status code
+ * @param message the status message
+ * @param body    the response body
+ */
+@Immutable
+public record RscpDTO<T>(@NonNull RscpStatus status, @Nullable String message, @Nullable T body) {
+}
```
Comment (Iwona007) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/dto/RscpDTO.java:18]: Co znaczy skrót Rscp?

**Comment by dawciobiel** [comment]
Please: Extend the documentation for RscpDTO, RscpStatus and within these classes describe what the RSCP abbreviation means.

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/dto/RscpDTO.java (hunk 1 above)
Comment (dawciobiel) [reply src/main/java/pl/byczazagroda/trackexpensesappbackend/dto/RscpDTO.java:18]: **R**esponse **S**ervice to **C**ontroller **P**rotocol.

**Comment by Iwona007** [comment]
We do not do this, that way

**Pull request closed without merging**